Support deployment from a private repo.

etam opened the pull request.

etam committed:
Support deployment from a private repo.

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+      local_action:
+        module: tempfile
+        state: directory
+        suffix: "_ar_repo"
```
please make this configurable and set a different default

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+        module: tempfile
+        state: directory
+        suffix: "_ar_repo"
+      register: ar_local_repo_path
```
please change the name to something else (not beginning with ar)

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+
+    - name: create git bundle
+      local_action:
+        module: shell git bundle create ../golem_requestor.bundle {{ golem_source_version|quote }} $(git describe --tags --abbrev=0) --
```
please change the name of the bundle

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+      local_action:
+        module: shell git bundle create ../golem_requestor.bundle {{ golem_source_version|quote }} $(git describe --tags --abbrev=0) --
+        args:
+          chdir: "{{ ar_local_repo_path.path }}/golem_git"
```
again, please change the prefix

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+    - name: transfer bundle to remote
+      become: yes
+      copy:
+        src: "{{ ar_local_repo_path.path }}/golem_requestor.bundle"
```
please change the prefix

ktomala replied:
and also bundle name

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+      become: yes
+      copy:
+        src: "{{ ar_local_repo_path.path }}/golem_requestor.bundle"
+        dest: "{{ golem_source_path }}/golem_requestor.bundle"
```
please change bundle name

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+    - name: clone git source from bundle
+      become: yes
+      git:
+        repo: "{{ golem_source_path }}/golem_requestor.bundle"
```
please change bundle name

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+
+  always:
+    - name: delete temporary directory with local repo clone
+      when: ar_local_repo_path is defined
```
please change the prefix

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+      when: ar_local_repo_path is defined
+      local_action:
+        module: file
+        path: "{{ ar_local_repo_path.path }}"
```
please change prefix

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+    - name: delete remote bundle file
+      become: yes
+      file:
+        path: "{{ golem_source_path }}/golem_requestor.bundle"
```
please change bundle name

ktomala commented on tasks/main.yml:
```diff
@@ -74,11 +74,87 @@
+      local_action:
+        module: git
+        repo: "{{ golem_source_repo }}"
+        dest: "{{ ar_local_repo_path.path }}/golem_git"
```
please change prefix

etam committed:
show what actually changed

ktomala reviewed (approved):
LGTM

etam committed:
move tasks using local clone to a separate file

This is required to avoid overwriting registered variables.
https://stackoverflow.com/q/34621799/952263
